packages/interactivity-router/src/index.js

```diff
@@ -117,21 +126,45 @@ export const { state, actions } = store( 'core/router', {
 				setTimeout( resolve, options.timeout ?? 10000 )
 			);
 
+			// Don't update the navigation status immediately, wait 400 ms.
+			const timeout = setTimeout( () => {
+				if ( navigatingTo === href ) {
+					navigation.hasStarted = true;
+					navigation.hasFinished = false;
+					navigation.message = navigation.texts.loading;
+				}
+			}, 400 );
+
 			const page = yield Promise.race( [
 				pages.get( url ),
 				timeoutPromise,
 			] );
 
+			// Dismiss loading message if it hasn't been added yet.
+			clearTimeout( timeout );
+
 			// Once the page is fetched, the destination URL could have changed
 			// (e.g., by clicking another link in the meantime). If so, bail
 			// out, and let the newer execution to update the HTML.
 			if ( navigatingTo !== href ) return;
 
+			navigation.hasStarted = false;
+			navigation.hasFinished = true;
```

Comment:
As these are going to be public APIs, it's a bit weird to me that `hasFinished` starts as `false` when the initial page loads, but then remains as `true` when the next page has loaded.

I don't have any suggestion on how to improve that. It's just a thought.

Reply:
I've finally left them as they are. I don't have any suggestions either. 🤷

Reply:
It's fine. Thanks.